var/spack/repos/builtin/packages/dftbplus/package.py

```diff
@@ -3,85 +3,142 @@
+    version(
+        "19.1",
+        sha256="4d07f5c6102f06999d8cfdb1d17f5b59f9f2b804697f14b3bc562e3ea094b8a8",
+        deprecated=True,
     )
 
-    variant("mpi", default=True, description="Build an MPI-paralelised version of the code.")
+    variant(
+        "build_type",
+        default="Release",
+        description="The build type to build",
+        values=("Debug", "Release", "RelWithDebInfo"),
+    )
 
     variant(
-        "gpu",
+        "api",
+        default=True,
+        description="Whether public API should be included and the DFTB+ library installed.",
+    )
+    variant(
+        "arpack",
         default=False,
-        description="Use the MAGMA library " "for GPU accelerated computation",
+        description="Whether the ARPACK library should be included (needed for TD-DFTB).",
+    )
+    variant(
+        "chimes",
+        default=False,
+        description="Whether repulsive corrections" "via the ChIMES library should be enabled.",
     )
-
     variant(
         "elsi",
         default=False,
         description="Use the ELSI library for large scale systems. "
         "Only has any effect if you build with '+mpi'",
     )
-
+    variant(
+        "gpu",
+        default=False,
+        description="Use the MAGMA library " "for GPU accelerated computation",
+    )
+    variant(
+        "mbd",
+        default=False,
+        description="Whether DFTB+ should be built with many-body-dispersion support.",
+    )
+    variant("mpi", default=False, description="Whether DFTB+ should support MPI-parallelism.")
+    variant(
+        "omp", default=True, description="Whether OpenMP thread parallisation should be enabled."
+    )
+    variant(
+        "plumed",
+        default=False,
+        description="Whether metadynamics via the PLUMED2 library should be allowed.",
+    )
+    variant("poisson", default=False, description="Whether the Poisson-solver should be included.")
+    variant(
+        "python",
+        default=False,
+        description="Whether the Python components of DFTB+ should be tested and installed.",
+    )
+    variant(
+        "sdftd3",
+        default=False,
+        description="Whether the s-dftd3 library should be included",
+    )
     variant(
         "sockets",
         default=False,
         description="Whether the socket library " "(external control) should be linked",
     )
-
-    variant("arpack", default=False, description="Use ARPACK for excited state DFTB functionality")
-
     variant(
         "transport",
         default=False,
         description="Whether transport via libNEGF should be included. "
         "Only affects parallel build. "
         "(serial version is built without libNEGF/transport)",
     )
+    variant(
+        "tblite", default=False, description="Whether xTB support should be included via tblite."
+    )
 
+    variant("shared", default=False, description="Most often for the Python wrappers.")
```

Comment:
The variant `-shared` needs to conflict with `+python` based on the description.